doc/development.rst

```diff
@@ -153,16 +153,22 @@ corresponding ``.hh`` files for C++ headers.
 
 Thus we have the following rules:
 
-- ``.hh`` is for C++ code compatible with host compilers. It may declare Thrust
-  objects, since thrust type declarations are compatible with the host
-  compiler. It can also use host/device keywords if it includes the CUDA
-  runtime API or hides the keywords with macros.
+- ``.hh`` is for C++ header code compatible with host compilers. The code in
+  this file can be compatible with device code if it uses the
+  ``CELER_FUNCTION`` family of macros defined in ``base/Macros.hh``.
 - ``.cc`` is for C++ code that will invariably be compiled by the host
   compiler.
-- ``.cu`` is for ``__global__`` kernels and functions that launch them
+- ``.cu`` is for ``__global__`` kernels and functions that launch them,
+  including code that initializes device memory.
 - ``.cuh`` is for header files that require compilation by NVCC: contain
   ``__device __``-only code or include CUDA directives without ``#include
   <cuda_runtime_api.h>``.
+- ``.cuda.hh`` and ``.cuda.cc`` require CUDA to be enabled and use CUDA runtime
+  libraries and headers, but they do not execute any device code and thus can
+  be built by a host compiler.  Memory-management code is a good example of
+  this.
+- ``.mpi.cc`` and ``.serial.cc`` code requires MPI to be enabled or disabled,
+  respectively.
 
 Ideally we could keep ``thrust`` includes isolated from ``.cu`` code to speed
 compilation and to potentially allow host testing of ``__device__``--annotated
```

Comment:
Any code including thrust requires the NVCC compiler

Reply:
Only thrust calls that invoke *kernels* need the NVCC compiler -- the rest of thrust just requires the CUDA C/C++ API (which it includes automatically). In practice though, since device vector constructors, `sort`, and all the other cuda-related goodies require kernel launches, you're effectively correct. I will update the documentation here.